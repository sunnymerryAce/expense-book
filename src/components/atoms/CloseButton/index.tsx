import React from "react";
import IconButton from "@material-ui/core/IconButton";
import CloseIcon from "@material-ui/icons/Close";
interface CloseButtonProps {
  handleClick: (props: unknown) => unknown;
}

const CloseButton: React.FC<CloseButtonProps> = (props) => {
  return (
    <IconButton
      onClick={props.handleClick}
      color="primary"
      className="CloseButton"
    >
      <CloseIcon />
    </IconButton>
  );
};

export default CloseButton;
