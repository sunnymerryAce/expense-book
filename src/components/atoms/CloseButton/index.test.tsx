import React from "react";
import CloseButton from "./";
import { shallow } from "enzyme";

describe("<CloseButton />", () => {
  let container = null;
  const handleClickFunction = jest.fn();

  beforeEach(() => {
    container = shallow(<CloseButton handleClick={handleClickFunction} />);
  });

  afterEach(() => {
    container.unmount();
    container = null;
  });

  test("should match the snapshot", () => {
    expect(container.html()).toMatchSnapshot();
  });

  test("should have proper props", () => {
    expect(container.find(".CloseButton").props()).toMatchObject({
      onClick: handleClickFunction,
    });
  });

  test("click event handler should be triggered", () => {
    container.simulate("click");
    expect(handleClickFunction).toBeCalled();
  });
});
