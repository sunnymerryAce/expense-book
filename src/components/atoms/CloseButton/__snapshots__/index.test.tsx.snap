// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CloseButton /> should match the snapshot 1`] = `"<button class=\\"MuiButtonBase-root MuiIconButton-root CloseButton MuiIconButton-colorPrimary\\" tabindex=\\"0\\" type=\\"button\\"><span class=\\"MuiIconButton-label\\"><svg class=\\"MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\"><path d=\\"M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z\\"></path></svg></span></button>"`;
